internal/nodes/dequeue_ingestion.go

```diff
@@ -175,6 +180,10 @@ func DequeueIngestion(key string, fullsync bool) {
 
 		for _, svcNSNameKey := range svcNames {
 			svcName := strings.Split(svcNSNameKey, "/")[1]
+			if svcObj, _ := utils.GetInformers().ServiceInformer.Lister().Services(namespace).Get(svcName); !lib.ValidateSvcforClass(key, svcObj) {
```

Comment:
@saurabhvaidvmware: If we look at overall flow, `handleL4SharedVipService` and `handleL4Service` are functions which deals with LB type services. These functions are called from different points (like node, l4Rule etc). You have added guard conditions in these functions already.
Should we take out such type of validation as that will be validated in above function by logic added.?

Reply:
@akshayhavile These 2 functions are being called in various places, all of which are not covered by my code changes. For eg, Pod objects and node objects ( `handleLBSvcUpdateForPod` calls `HandleL4Service` ). 

- I've placed checks in ingestion of basic LBsvcs, but other objects like endpoints are still being ingested and can lead to VS creation for LB Svcs, it is not possible to stop their ingestion due to association with other Service Types. So, I added validation check in final step  (HandleL4 functions)

-  Additionally if some new object type is created in the future that affects LBSvcs, it will also call on these 2 functions. With validation checks in the handleL4 functions itself, we will not have to worry about adding validation in between during ingestion or dequeuing. 
-  I also added these as redundancy checks,  to make sure that nothing leaks through.

Please let me know if I made some mistake in my logic somewhere, I'll go ahead and change the code accordingly.

Reply:
All L4 service creation flows lands on functions `handleLBSvcUpdateForPod` and `handleL4Service`. So as per your say ( taking into consideration future flows and different ingestions point) if we just keep validation in these functions and remove from other places, what will be impact? We will keep validation of L4 Objects in during event handler addition for L4 service, but from remaining places like this we can remove the call.